src/main/java/com/hivemq/mqtt/services/PublishPollServiceImpl.java

```diff
@@ -181,12 +181,19 @@ public void onSuccess(final ImmutableList<PUBLISH> publishes) {
                 for (int i = usedIds; i < messageIds.length(); i++) {
                     messageIDPool.returnId(messageIds.get(i));
                 }
-
+                final List<PublishWithFuture> publishesToSend = new ArrayList<>();
                 final AtomicInteger inFlightMessages = inFlightMessageCount(channel);
                 for (final PUBLISH publish : publishes) {
                     inFlightMessages.incrementAndGet();
                     try {
-                        sendOutPublish(publish, false, channel, client, messageIDPool, client);
+                        payloadPersistence.add(publish.getPayload(), 1, publish.getPublishId());
```

Comment:
why do we store the payload in the payload persistence here?

Reply:
its a relict, ill remove it